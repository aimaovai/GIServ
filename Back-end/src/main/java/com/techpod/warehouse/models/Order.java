package com.techpod.warehouse.models;

import javax.persistence.*;

@Entity
@Table(name = "orders")
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public int id;
    public String product;
    public int quantity;
    public int value;
    public String shippingdetails;
    public boolean shipped;

    @Override
    public String toString() {
        return "Order{" +
                "id=" + id +
                ", product='" + product + '\'' +
                ", quantity=" + quantity +
                ", value=" + value +
                ", shippingdetails='" + shippingdetails + '\'' +
                ", shipped=" + shipped +
                '}';
    }
}
