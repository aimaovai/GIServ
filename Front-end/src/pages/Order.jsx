import React, { useEffect, useState } from "react";
import api from "../api";
import Layout from "../components/Layout";
import handleForm from "../helper";
import { checkLoggedIn } from "../helper/AuthHelper";

export default function Order() {
  const [Order, setOrder] = useState(null);
  const [Products, setProducts] = useState(null);

  const [form, setForm] = useState({
    name: "",
    vendor: "",
    quantity: "",
    value: 0,
    shippingdetails: "",
    shipped: false,
  });

  useEffect(() => {
    checkLoggedIn();

    getAllOrder();
    getAllProducts();
  }, []);

  const newOrder = () => {
    console.log(form);

    if (
      form.vendor === "" ||
      form.shippingdetails === "" ||
      form.quantity === "" ||
      form.value === ""
    ) {
      alert("Fields Cannot Be Empty");
      return;
    }

    api.addOrder(form).then((res) => {
      getAllOrder();
      alert("Successfully Order Added");
      setForm({
        name: "",
        vendor: "",
        stock: "",
        product:"",
        value: 0,
        shippingdetails: "",
        shipped: false,
      });
    });
  };

  const [stats, setStats] = useState({
    total: 0,
    free: 0,
    notfree: 0,
  });

  const getAllOrder = async () => {
    let Order = await api.getOrders();
    console.log(Order);

    let free = 0,
      notfree = 0;

    Order.map((order) => {
      order.shipped ? (free += 1) : (notfree += 1);
    });

    setStats({
      total: Order.length,
      free,
      notfree,
    });

    setOrder(Order);
  };

  const getAllProducts = async () => {
    let Products = await api.getProducts();
    console.log(Products);
    setProducts([{ name: "Select Product" }, ...Products]);
  };

  const removeOrder = (Order) => {
    api.deleteOrder(Order).then((res) => {
      getAllOrder();
      alert("Successfully Order Removed");
    });
  };

  const handleOrder = (e) => {
    if (e.target.value === "Select Product") {
      setForm({ ...form, value: 0 });
      return;
    }

    setForm(handleForm(e, form));

    Products.map((product) => {
      if (e.target.value === product.name) {
        setForm({
          ...form,
          product: e.target.value,
          vendor: product.vendor,
          value: form.quantity * product.price,
        });
      }
    });
  };

  const changeForm = (e) => {
    Products.map((product) => {
      if (form.product === product.name) {
        setForm({
          ...form,
          value: e.target.value * product.price,
        });
      }
    });
  }

  const update = (data) => {
    let tempData = {
      ...data,
      shipped: !data.shipped,
    };
    api.addOrder(tempData).then((res) => {
      getAllOrder();
    });
  };

  return (
    <Layout>
      <div data-testid="order" className="flex justify-around">
        <div className="h-20 w-36 bg-blue-50 rounded-lg text-blue-500 text-center p-2">
          <p className="text-4xl">{stats.total}</p>
          <p className="text-sm">Total</p>
        </div>
        <div className="h-20 w-36 bg-green-50 rounded-lg text-green-500 text-center p-2">
          <p className="text-4xl">{stats.free}</p>
          <p className="text-sm">Shipped</p>
        </div>
        <div className="h-20 w-36 bg-red-50 rounded-lg text-red-500 text-center p-2">
          <p className="text-4xl">{stats.notfree}</p>
          <p className="text-sm">Not Shipped</p>
        </div>
      </div>

      <div data-testid="orderinsert" className="flex gap-2 pt-5 pb-2 justify-center">
        <input
          type="text"
          placeholder="quantity"
          name="quantity"
          className="h-10 outline-none bg-slate-100 text-center rounded-md"
          onChange={(e) => {
            setForm(handleForm(e, form))
            changeForm(e)
          }}
        />
        <select
          className="outline-none bg-slate-100 rounded-md px-3"
          name="product"
          onChange={(e) => {
            handleOrder(e);
          }}
        >
          {Products &&
            Products.map((product) => {
              return <option value={product.name}>{product.name}</option>;
            })}
        </select>
        <input
          type="text"
          placeholder="total"
          name="value"
          value={form.value}
          disabled
          className="h-10 outline-none bg-slate-100 text-center rounded-md"
        />

        <input
          type="text"
          placeholder="shippingdetails"
          name="shippingdetails"
          className="h-10 outline-none bg-slate-100 text-center rounded-md"
          onChange={(e) => setForm(handleForm(e, form))}
        />

        <button
          data-testid="insert-order-button"
          className="bg-blue-600 h-10 rounded-md text-white text-sm text-center px-3"
          onClick={() => newOrder()}
        >
          Add Order
        </button>
      </div>

      <table data-testid="orders" className="w-full text-center my-6 rounded-md bg-slate-50 ">
        <thead>
          <tr>
            <th>Id</th>
            <th>Item</th>
            <th>Quantity</th>
            <th>Value</th>
            <th>Address</th>
            <th>Shipped</th>
            <th>Actions</th>
          </tr>
        </thead>
        <tbody>
          {Order &&
            Order.map((order) => {
              return (
                <tr>
                  <td>{order.id}</td>
                  <td>{order.product}</td>
                  <td>{order.quantity}</td>
                  <td>{order.value}</td>
                  <td>{order.shippingdetails}</td>
                  <td>
                    {order.shipped ? "Yes" : "No"}
                    <button
                      data-testid="toggle-button"
                      onClick={() => update(order)}
                      className="text-sm bg-green-300 p-1 rounded-md m-1"
                    >
                      Change
                    </button>
                  </td>
                  <td>
                    <button
                      className="text-sm bg-red-50 text-red-500 px-2 py-1 rounded-md"
                      onClick={() => removeOrder(order.id)}
                    >
                      Remove
                    </button>
                  </td>
                </tr>
              );
            })}
        </tbody>
      </table>
    </Layout>
  );
}
