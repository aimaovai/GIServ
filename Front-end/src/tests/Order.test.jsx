/* 
  - Test suite id: TS_05
  - Test suite written by: Aima Ovai
  - Test suite written on: April 7, 2022
  - Test suite last updated on: April 28, 2022
  - Test Description: This test suite is written to test the fucntionality
    of the Order webpage. It tests that the page renders properly, 
    and that the order details are being retreived from the database. 
*/

import React from "react";
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, mount, configure } from "enzyme";
import { render, screen, waitFor,fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

import Order from '../pages/Order';

// jest.mock('../helper/AuthHelper');

// const func = require('../helper/AuthHelper');

Enzyme.configure({adapter: new Adapter()});

describe('GIServ Order Component', () => {

  // Test case: TC_12
  test('user is logged in', () => {
    try {
      const loggedin = jest.fn().mockReturnValueOnce("loggedin");
      Storage.prototype.setItem = jest.fn(() => {
        console.log("isUserLoggedin")
      });

      checkLoggedIn.setItem("isUserLoggedin", loggedin);
      } catch (error) {
        console.log('User not logged in.')
      }
  })
  
  // Test case: TC_13
  it("Order renders without crashing", async () => {
    render(<Order />)

    // await expect(render(<Order />)).toMatchSnapshot()
      const orderItem = await waitFor(() => screen.getByTestId('order'));
      expect(orderItem).toBeInTheDocument();
  });

  // Test case: TC_14
  test('it displays a table for Orders', async () => {
    
      render(<Order />);

      const orderList = await waitFor(() => screen.getByTestId('orders'));
      expect(orderList).toBeInTheDocument();
  })

  // Test case: TC_15
  test('it retrieves data', async () => {
   
      render(<Order />);

      const orderList = await waitFor(() => screen.findAllByTestId('orders'));
      expect(orderList).toHaveLength(1);
  })

  // Test case: TC_16
  test('test Order insert button on click', () => {
    const { getByTestId } = render(<Order />);
    fireEvent.click(getByTestId('insert-order-button'));
    expect(getByTestId('orderinsert')).toHaveTextContent('Add Order');
  });

  // Test case: TC_17
  test('test that Order status toggle button exists', () => {

      const wrapper = mount(<Order />);
      const table = wrapper.find('table');
      const row = table.find('tr');
      const togleButton = table.find('toggle-button');

      expect(table).toHaveLength(1);
      expect(row).toHaveLength(1);
      expect(togleButton).not.toBeNull();
  });
})