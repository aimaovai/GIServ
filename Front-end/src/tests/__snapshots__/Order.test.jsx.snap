// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIServ Order Component Order page renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="h-full bg-gradient-to-b from-red-500 to-purple-700 p-8"
      >
        <div
          class="flex justify-around"
        >
          <div
            class="h-20 w-36 bg-blue-50 rounded-lg text-blue-500 text-center p-2"
          >
            <p
              class="text-4xl"
            >
              0
            </p>
            <p
              class="text-sm"
            >
              Total
            </p>
          </div>
          <div
            class="h-20 w-36 bg-green-50 rounded-lg text-green-500 text-center p-2"
          >
            <p
              class="text-4xl"
            >
              0
            </p>
            <p
              class="text-sm"
            >
              Shipped
            </p>
          </div>
          <div
            class="h-20 w-36 bg-red-50 rounded-lg text-red-500 text-center p-2"
          >
            <p
              class="text-4xl"
            >
              0
            </p>
            <p
              class="text-sm"
            >
              Not Shipped
            </p>
          </div>
        </div>
        <div
          class="flex gap-2 pt-5 pb-2 justify-center"
        >
          <input
            class="h-10 outline-none bg-slate-100 text-center rounded-md"
            name="quantity"
            placeholder="quantity"
            type="text"
          />
          <select
            class="outline-none bg-slate-100 rounded-md px-3"
            name="product"
          />
          <input
            class="h-10 outline-none bg-slate-100 text-center rounded-md"
            disabled=""
            name="value"
            placeholder="total"
            type="text"
            value="0"
          />
          <input
            class="h-10 outline-none bg-slate-100 text-center rounded-md"
            name="shippingdetails"
            placeholder="shippingdetails"
            type="text"
          />
          <button
            class="bg-blue-600 h-10 rounded-md text-white text-sm text-center px-3"
          >
            Add Order
          </button>
        </div>
        <table
          class="w-full text-center my-6 rounded-md bg-slate-50 "
        >
          <thead>
            <tr>
              <th>
                Id
              </th>
              <th>
                Item
              </th>
              <th>
                Quantity
              </th>
              <th>
                Value
              </th>
              <th>
                Address
              </th>
              <th>
                Shipped
              </th>
              <th>
                Actions
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
    <div>
      <div
        class="h-full bg-gradient-to-b from-red-500 to-purple-700 p-8"
      >
        <div
          class="flex justify-around"
        >
          <div
            class="h-20 w-36 bg-blue-50 rounded-lg text-blue-500 text-center p-2"
          >
            <p
              class="text-4xl"
            >
              0
            </p>
            <p
              class="text-sm"
            >
              Total
            </p>
          </div>
          <div
            class="h-20 w-36 bg-green-50 rounded-lg text-green-500 text-center p-2"
          >
            <p
              class="text-4xl"
            >
              0
            </p>
            <p
              class="text-sm"
            >
              Shipped
            </p>
          </div>
          <div
            class="h-20 w-36 bg-red-50 rounded-lg text-red-500 text-center p-2"
          >
            <p
              class="text-4xl"
            >
              0
            </p>
            <p
              class="text-sm"
            >
              Not Shipped
            </p>
          </div>
        </div>
        <div
          class="flex gap-2 pt-5 pb-2 justify-center"
        >
          <input
            class="h-10 outline-none bg-slate-100 text-center rounded-md"
            name="quantity"
            placeholder="quantity"
            type="text"
          />
          <select
            class="outline-none bg-slate-100 rounded-md px-3"
            name="product"
          />
          <input
            class="h-10 outline-none bg-slate-100 text-center rounded-md"
            disabled=""
            name="value"
            placeholder="total"
            type="text"
            value="0"
          />
          <input
            class="h-10 outline-none bg-slate-100 text-center rounded-md"
            name="shippingdetails"
            placeholder="shippingdetails"
            type="text"
          />
          <button
            class="bg-blue-600 h-10 rounded-md text-white text-sm text-center px-3"
          >
            Add Order
          </button>
        </div>
        <table
          class="w-full text-center my-6 rounded-md bg-slate-50 "
        >
          <thead>
            <tr>
              <th>
                Id
              </th>
              <th>
                Item
              </th>
              <th>
                Quantity
              </th>
              <th>
                Value
              </th>
              <th>
                Address
              </th>
              <th>
                Shipped
              </th>
              <th>
                Actions
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="h-full bg-gradient-to-b from-red-500 to-purple-700 p-8"
    >
      <div
        class="flex justify-around"
      >
        <div
          class="h-20 w-36 bg-blue-50 rounded-lg text-blue-500 text-center p-2"
        >
          <p
            class="text-4xl"
          >
            0
          </p>
          <p
            class="text-sm"
          >
            Total
          </p>
        </div>
        <div
          class="h-20 w-36 bg-green-50 rounded-lg text-green-500 text-center p-2"
        >
          <p
            class="text-4xl"
          >
            0
          </p>
          <p
            class="text-sm"
          >
            Shipped
          </p>
        </div>
        <div
          class="h-20 w-36 bg-red-50 rounded-lg text-red-500 text-center p-2"
        >
          <p
            class="text-4xl"
          >
            0
          </p>
          <p
            class="text-sm"
          >
            Not Shipped
          </p>
        </div>
      </div>
      <div
        class="flex gap-2 pt-5 pb-2 justify-center"
      >
        <input
          class="h-10 outline-none bg-slate-100 text-center rounded-md"
          name="quantity"
          placeholder="quantity"
          type="text"
        />
        <select
          class="outline-none bg-slate-100 rounded-md px-3"
          name="product"
        />
        <input
          class="h-10 outline-none bg-slate-100 text-center rounded-md"
          disabled=""
          name="value"
          placeholder="total"
          type="text"
          value="0"
        />
        <input
          class="h-10 outline-none bg-slate-100 text-center rounded-md"
          name="shippingdetails"
          placeholder="shippingdetails"
          type="text"
        />
        <button
          class="bg-blue-600 h-10 rounded-md text-white text-sm text-center px-3"
        >
          Add Order
        </button>
      </div>
      <table
        class="w-full text-center my-6 rounded-md bg-slate-50 "
      >
        <thead>
          <tr>
            <th>
              Id
            </th>
            <th>
              Item
            </th>
            <th>
              Quantity
            </th>
            <th>
              Value
            </th>
            <th>
              Address
            </th>
            <th>
              Shipped
            </th>
            <th>
              Actions
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`GIServ Order Component Order renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="h-full bg-gradient-to-b from-red-500 to-purple-700 p-8"
      >
        <div
          class="flex justify-around"
        >
          <div
            class="h-20 w-36 bg-blue-50 rounded-lg text-blue-500 text-center p-2"
          >
            <p
              class="text-4xl"
            >
              0
            </p>
            <p
              class="text-sm"
            >
              Total
            </p>
          </div>
          <div
            class="h-20 w-36 bg-green-50 rounded-lg text-green-500 text-center p-2"
          >
            <p
              class="text-4xl"
            >
              0
            </p>
            <p
              class="text-sm"
            >
              Shipped
            </p>
          </div>
          <div
            class="h-20 w-36 bg-red-50 rounded-lg text-red-500 text-center p-2"
          >
            <p
              class="text-4xl"
            >
              0
            </p>
            <p
              class="text-sm"
            >
              Not Shipped
            </p>
          </div>
        </div>
        <div
          class="flex gap-2 pt-5 pb-2 justify-center"
        >
          <input
            class="h-10 outline-none bg-slate-100 text-center rounded-md"
            name="quantity"
            placeholder="quantity"
            type="text"
          />
          <select
            class="outline-none bg-slate-100 rounded-md px-3"
            name="product"
          />
          <input
            class="h-10 outline-none bg-slate-100 text-center rounded-md"
            disabled=""
            name="value"
            placeholder="total"
            type="text"
            value="0"
          />
          <input
            class="h-10 outline-none bg-slate-100 text-center rounded-md"
            name="shippingdetails"
            placeholder="shippingdetails"
            type="text"
          />
          <button
            class="bg-blue-600 h-10 rounded-md text-white text-sm text-center px-3"
          >
            Add Order
          </button>
        </div>
        <table
          class="w-full text-center my-6 rounded-md bg-slate-50 "
        >
          <thead>
            <tr>
              <th>
                Id
              </th>
              <th>
                Item
              </th>
              <th>
                Quantity
              </th>
              <th>
                Value
              </th>
              <th>
                Address
              </th>
              <th>
                Shipped
              </th>
              <th>
                Actions
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
    <div>
      <div
        class="h-full bg-gradient-to-b from-red-500 to-purple-700 p-8"
      >
        <div
          class="flex justify-around"
        >
          <div
            class="h-20 w-36 bg-blue-50 rounded-lg text-blue-500 text-center p-2"
          >
            <p
              class="text-4xl"
            >
              0
            </p>
            <p
              class="text-sm"
            >
              Total
            </p>
          </div>
          <div
            class="h-20 w-36 bg-green-50 rounded-lg text-green-500 text-center p-2"
          >
            <p
              class="text-4xl"
            >
              0
            </p>
            <p
              class="text-sm"
            >
              Shipped
            </p>
          </div>
          <div
            class="h-20 w-36 bg-red-50 rounded-lg text-red-500 text-center p-2"
          >
            <p
              class="text-4xl"
            >
              0
            </p>
            <p
              class="text-sm"
            >
              Not Shipped
            </p>
          </div>
        </div>
        <div
          class="flex gap-2 pt-5 pb-2 justify-center"
        >
          <input
            class="h-10 outline-none bg-slate-100 text-center rounded-md"
            name="quantity"
            placeholder="quantity"
            type="text"
          />
          <select
            class="outline-none bg-slate-100 rounded-md px-3"
            name="product"
          />
          <input
            class="h-10 outline-none bg-slate-100 text-center rounded-md"
            disabled=""
            name="value"
            placeholder="total"
            type="text"
            value="0"
          />
          <input
            class="h-10 outline-none bg-slate-100 text-center rounded-md"
            name="shippingdetails"
            placeholder="shippingdetails"
            type="text"
          />
          <button
            class="bg-blue-600 h-10 rounded-md text-white text-sm text-center px-3"
          >
            Add Order
          </button>
        </div>
        <table
          class="w-full text-center my-6 rounded-md bg-slate-50 "
        >
          <thead>
            <tr>
              <th>
                Id
              </th>
              <th>
                Item
              </th>
              <th>
                Quantity
              </th>
              <th>
                Value
              </th>
              <th>
                Address
              </th>
              <th>
                Shipped
              </th>
              <th>
                Actions
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="h-full bg-gradient-to-b from-red-500 to-purple-700 p-8"
    >
      <div
        class="flex justify-around"
      >
        <div
          class="h-20 w-36 bg-blue-50 rounded-lg text-blue-500 text-center p-2"
        >
          <p
            class="text-4xl"
          >
            0
          </p>
          <p
            class="text-sm"
          >
            Total
          </p>
        </div>
        <div
          class="h-20 w-36 bg-green-50 rounded-lg text-green-500 text-center p-2"
        >
          <p
            class="text-4xl"
          >
            0
          </p>
          <p
            class="text-sm"
          >
            Shipped
          </p>
        </div>
        <div
          class="h-20 w-36 bg-red-50 rounded-lg text-red-500 text-center p-2"
        >
          <p
            class="text-4xl"
          >
            0
          </p>
          <p
            class="text-sm"
          >
            Not Shipped
          </p>
        </div>
      </div>
      <div
        class="flex gap-2 pt-5 pb-2 justify-center"
      >
        <input
          class="h-10 outline-none bg-slate-100 text-center rounded-md"
          name="quantity"
          placeholder="quantity"
          type="text"
        />
        <select
          class="outline-none bg-slate-100 rounded-md px-3"
          name="product"
        />
        <input
          class="h-10 outline-none bg-slate-100 text-center rounded-md"
          disabled=""
          name="value"
          placeholder="total"
          type="text"
          value="0"
        />
        <input
          class="h-10 outline-none bg-slate-100 text-center rounded-md"
          name="shippingdetails"
          placeholder="shippingdetails"
          type="text"
        />
        <button
          class="bg-blue-600 h-10 rounded-md text-white text-sm text-center px-3"
        >
          Add Order
        </button>
      </div>
      <table
        class="w-full text-center my-6 rounded-md bg-slate-50 "
      >
        <thead>
          <tr>
            <th>
              Id
            </th>
            <th>
              Item
            </th>
            <th>
              Quantity
            </th>
            <th>
              Value
            </th>
            <th>
              Address
            </th>
            <th>
              Shipped
            </th>
            <th>
              Actions
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
